.github/workflows/build.yml

```diff
@@ -30,112 +34,102 @@ jobs:
+        sudo apt-get -y remove --purge software-properties-common
+        sudo apt-get -y autoremove
 
     - name: Select optimal cabal version
       run: |
         case "$OS" in
-          Windows_NT)   echo "CABAL_VERSION=3.2.0.0"      >> $GITHUB_ENV;;
+          Windows_NT)   echo "CABAL_VERSION=3.4.0.0-rc5"  >> $GITHUB_ENV;;
           *)            echo "CABAL_VERSION=3.4.0.0-rc4"  >> $GITHUB_ENV;;
         esac
 
-    - uses: actions/setup-haskell@v1
+    - uses: haskell/actions/setup@v1
       id: setup-haskell
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ env.CABAL_VERSION }}
 
-    - name: Patch GHC 8.10.2 linker
-      if: matrix.os == 'windows-latest' && matrix.ghc == '8.10.2'
+    - name: Haskell versions
       run: |
-        sed -i \
-          's|C:/GitLabRunner/builds/2WeHDSFP/0/ghc/ghc/inplace/mingw/bin/ld.exe|C:/ProgramData/chocolatey/lib/ghc.8.10.2/tools/ghc-8.10.2/mingw/bin/ld.exe|g' \
-          C:/ProgramData/chocolatey/lib/ghc.8.10.2/tools/ghc-8.10.2/lib/settings
-
-    - name: Install build environment
-      if: matrix.os == 'ubuntu-latest'
-      run: |
-        sudo apt-get update
-        sudo apt-get -y install libsodium23 libsodium-dev
-        sudo apt-get -y install libsystemd0 libsystemd-dev
-        sudo apt-get -y remove --purge software-properties-common
-        sudo apt-get -y autoremove
+        ghc --version
+        cabal --version
 
     - name: Set up temp directory
+      env:
+        RUNNER_TEMP: ${{ runner.temp }}
       run: |
-        echo 'TMPDIR=${{ runner.temp }}'  >> $GITHUB_ENV
-        echo 'TMP=${{ runner.temp }}'     >> $GITHUB_ENV
+        echo "TMPDIR=$RUNNER_TEMP"  >> $GITHUB_ENV
+        echo "TMP=$RUNNER_TEMP"     >> $GITHUB_ENV
 
     # Cache cabal-store
     - name: cache cabal store
       uses: actions/cache@v2
       with:
         path: ${{ steps.setup-haskell.outputs.cabal-store }}
-        key: cabal-store-${{ matrix.ghc }}-${{ matrix.os }}
-    # Cache parts of dist-newstyle (C:\dist)
-    - name: cache buiddir [cache]
-      uses: actions/cache@v2
-      with:
-        path: C:\dist\cache
-        key: dist-cache-${{ matrix.ghc }}-${{ matrix.os }}
-    - name: cache buiddir [src]
-      uses: actions/cache@v2
-      with:
-        path: C:\dist\src
-        key: dist-src-${{ matrix.ghc }}-${{ matrix.os }}
-    - name: cache builddir [packagedb]
-      uses: actions/cache@v2
-      with:
-        path: C:\dist\packagedb
-        key: dist-packagedb-${{ matrix.ghc }}-${{ matrix.os }}
+        key: cabal-store-${{ env.CACHE_VERSION }}-${{ matrix.ghc }}-${{ matrix.os }}
 
     - name: update Hackage index
       run: cabal v2-update
 
-    # checkout repository
-    - name: checkout ouroboros-network repository
-      uses: actions/checkout@v2
-
     - name: Cabal Configure
       run: cabal --builddir="$CABAL_BUILDDIR" configure --enable-tests
 
     - name: Use cabal.project.local.windows
```

Comment:
Should be `Use cabal.project.local.ci.windows`, I think.

Reply:
Fixed.  Thanks!